File: languages/python/test-one.py
import mf2py
import sys
import json
if len(sys.argv) < 2:
    print('usage: '+ sys.argv[0]+ ' <inputfile> <outputfile>')
    sys.exit(1)

name = sys.argv[1]
base = 'http://example.com/'
if name.startswith("vendor/mf2/tests/tests/microformats-v2-unit/"):
    # This is a unit test; these use a different base URL
    base = 'http://example.test'

with open(name,'r') as file:
    p = mf2py.Parser(doc=file, url=base)
    res = json.loads(p.to_json())
    del res['debug']
    print(json.dumps(res))

File: languages/typescript/test-one.js
const { mf2 } = require('microformats-parser')

var infile = process.argv[2];

if (infile === undefined) {
  console.log("Usage: " + process.argv[1] + " <inputfile>");
  process.exit(1);
}

var base = 'http://example.com/';
if (infile.indexOf("/microformats-v2-unit/") > -1) {
    // This is a unit test; these use a different base URL
    base = 'http://example.test';
}

var fs = require("fs");

fs.readFile(infile, "utf8", function (err, data) {
    if (err) throw err;

    const parsed = mf2(data, {
        baseUrl: base,
    });
    console.log("%j", parsed);
});

